File: leetcode/tree/bt_level_order_traversal.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	if root == nil {
		return nil
	}
	var q []*TreeNode
	var res [][]int
	q = append(q, root)
	for len(q) != 0 {
		var inner []int
		size := len(q)
		for i := 0; i < size; i++ {
			tmp := q[0]
			q = q[1:]
			if tmp.Left != nil {
				q = append(q, tmp.Left)
			}

			if tmp.Right != nil {
				q = append(q, tmp.Right)
			}

			inner = append(inner, tmp.Val)
		}
		res = append(res, inner)
	}
	return res
}

func main() {
	root := &TreeNode{3, nil, nil}
	root.Left = &TreeNode{9, nil, nil}
	root.Right = &TreeNode{20, nil, nil}
	root.Right.Left = &TreeNode{15, nil, nil}
	root.Right.Right = &TreeNode{7, nil, nil}
	out := levelOrder(root)
	for i := 0; i < len(out); i++ {
		for j := 0; j < len(out[i]); j++ {
			fmt.Printf("%d ", out[i][j])
		}
		fmt.Printf("\n")
	}
}

File: leetcode/tree/bt_level_order_traversal.py
from typing import List

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return None
        res = []
        queue = []
        queue.append(root)
        while len(queue) != 0:
            inner = []
            for _ in range(len(queue)):
                tmp = queue.pop(0)
                if tmp.left:
                    queue.append(tmp.left)
                if tmp.right:
                    queue.append(tmp.right)
                inner.append(tmp.val)
            res.append(inner)
        return res

if __name__ == "__main__":
    root = TreeNode(3)
    root.left = TreeNode(9)
    root.right = TreeNode(20)
    root.right.left = TreeNode(15)
    root.right.right = TreeNode(7)
    out = Solution().levelOrder(root)
    print(out)